executor/mem_reader.go

```diff
@@ -90,15 +90,36 @@ func buildMemIndexReader(ctx context.Context, us *UnionScanExec, idxReader *Inde
 }
 
 func (m *memIndexReader) getMemRowsIter(ctx context.Context) (memRowsIter, error) {
-	data, err := m.getMemRows(ctx)
+	if m.keepOrder && m.table.GetPartitionInfo() != nil {
+		data, err := m.getMemRows(ctx)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		return &defaultRowsIter{data: data}, nil
+	}
+
+	kvIter, err := newTxnMemBufferIter(m.ctx, m.cacheTable, m.kvRanges, m.desc)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return &defaultRowsIter{data: data}, nil
+	tps := m.getTypes()
+	hdStatus := tablecodec.HandleDefault
+	if mysql.HasUnsignedFlag(tps[len(tps)-1].GetFlag()) {
```

Comment:
How does this work with PK(i int, u int unsigned) vs PK(u int unsigned, i int) ? I.e. can the handle be said to be either HandleIsUnsigned or HandleDefault in either of those cases?

Reply:
I don't change the code logic, it's here originally:
https://github.com/pingcap/tidb/pull/45786/files#diff-f7f3339231e4104abed7ddd0b9a3e12d2284761165df20d2141406104051767cL159-L164

Reply:
Cool, do you see any possible bugs due to this? (not related to this pr)